backend/addresses/serializers.py

```diff
@@ -0,0 +1,81 @@
+import re
+from rest_framework import serializers
+from .models import Address
+from accounts.models import CustomerProfile
+
+EGYPT_GOVERNORATES = [
+    "Cairo", "Alexandria", "Giza", "Aswan", "Luxor", "Suez", "Port Said", "Ismailia", "Dakahlia",
+    "Qalyubia", "Sharqia", "Beheira", "Gharbia", "Monufia", "Faiyum", "Beni Suef", "Minya", "Sohag",
+    "Qena", "Assiut", "Red Sea", "New Valley", "North Sinai", "South Sinai", "Matruh", "Damietta",
+    "Kafr El Sheikh"
+]
+
+class AddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Address
+        fields = '__all__'
+        read_only_fields = ['user']
+
+    def validate_postal_code(self, value):
+        if not re.match(r"^\d{5}$", value):
+            raise serializers.ValidationError("Postal code must be 5 digits.")
+        return value
+
+    def validate_city(self, value):
+        if not value.strip():
+            raise serializers.ValidationError("City is required.")
+        return value
+
+    def validate_governorate(self, value):
+        if value not in EGYPT_GOVERNORATES:
+            raise serializers.ValidationError("Invalid governorate.")
+        return value
+
+    def validate_country(self, value):
+        if value.strip().lower() != "egypt":
+            raise serializers.ValidationError("Country must be 'Egypt'.")
+        return value
+
```

Comment:
Add a generic validation to make sure that only users with `customer` role, could create an address with `is_default=True`.

Here is my suggestion:
```suggestion

    def validate(self, attrs):
      user = self.context["request"].user
      is_default = attrs.get("is_default", False)
      
      if is_default and user.role != "customer":
        raise serializers.ValidationError("Only customers can set a default shipping address.")
      return attrs
```

This ensures:
  - No role other than `"customer"` could submit an address with `"is_default": true` inside the request's body.
  - Gives a 400 response before any DB queries are made.
  - Adds another layer of protection through validation.